src/vault.rs

```diff
@@ -64,6 +67,12 @@ struct AppIdAuthRequest<'a> {
     user_id: &'a str,
 }
 
+#[derive(Serialize)]
+pub struct AuthKubernetesRequest {
+    pub jwt: String,
+    pub role: String
+}
```

Comment:
This should be able to just take string references like:
``` rust
#[derive(Serialize)]
pub struct AuthKubernetesRequest<'a> {
    pub jwt: &'a str,
    pub role: &'a str,
}
```

You'll with that change probably need to create an instance like:

``` rust
AuthKubernetesRequest { jwt: &jwt, role }
```

as you'll have an owned `String`, but it wants a `&str` reference (`role` will already be a `&str` reference if you follow my other comments).